src/frontend/src/lib/utils/tokens.utils.ts

```diff
@@ -76,15 +106,8 @@ export const pinTokensWithBalanceAtTop = ({
 	const [positiveBalances, nonPositiveBalances] = $tokens.reduce<[TokenUi[], TokenUi[]]>(
 		(acc, token) => {
 			const balance: BigNumber | undefined = $balances?.[token.id]?.data;
-			const exchangeRate: number | undefined = $exchanges?.[token.id]?.usd;
 
-			const usdBalance: number | undefined = nonNullish(exchangeRate)
-				? usdValue({
-						token,
-						balance,
-						exchangeRate
-					})
-				: undefined;
+			const usdBalance = calculateTokenUsdBalance({ token, $balances, $exchanges });
```

Comment:
just for readability, i would leave the type here too

`: number | undefined`

Reply:
done